### PR TITLE
Migrating core-data to Config V2

### PR DESCRIPTION
**Pull request by tsconn23**
#547 

The bulk of this PR is targeted at integrating the ConfigV2 scheme with the core-data service, however it also includes a significant bug fix to the previous core-command PR #541 that was uncovered during testing of the /api/v1/config endpoint. See below.

- Added /api/v1/config endpoint to return currently populated ConfigurationStruct in JSON format to both core-data and core-command. This has also been added to the core-data/command RAML.
- The above uncovered a bug in the previous approach that showed configuration information was never being pulled from Consul, but rather always held values assigned during initial load from file.
- In order to fetch information from Consul, one of the changes required the unfortunate elimination of the BaseConfig struct. This is because the consuldecoder library encountered a mismatch when pulling from Consul. When attempting to fetch Consul key config/V2/edgex-core-data/Clients, reflection on the service's ConfigurationStruct type with the BaseConfig included would require the above path to be config/V2/edgex-core-data/BaseConfig/Clients. I could have made this so in the TOML file, but that didn't seem like an attractive approach.
- Also fixed a bug in core/data/domain_events.go wherein the channel was closed after receiving the first event due to missing for{}
- Added explicit call to clear resources used by GorillaMUX when handlers complete per documentation. See core-data/command main.go startHttpServer()
- Core-command and core-data now utilize a Consul watcher through the imported consuldecoder package. This means that after a service bootstraps, you can change values in Consul and see them immediately reflected via that service's /api/v1/config endpoint.
- I did give some thought to how the initializeConfiguration() and listenForConfigChanges() functions in the init.go files might be restructured to avoid having to copy them from service to service. Unfortunately, there is coupling with each service's ConfigurationStruct type that I don't see a way to avoid at the moment. 

I look forward to your comments.
 
Signed-off-by: Trevor Conn <trevor_conn@dell.com>

### CODE REVIEW

**Comment by jpwhitemn** [comment]
LGTM Trevor.  Not approving to allow others to comment before it accidentally gets merged.  I like that you added the /api/v1/config endpoint.  This will help system management and @akramtexas greatly (guessing you guys coordinated on this).

**Comment by akramtexas** [comment]
Yep @jpwhitemn in connection with the /api/v1/config endpoint, @tsconn23 had reached out to me (Thanks, again, Trevor), that being the endpoint I had added to the notifications micro-service in the branch for (ongoing) MSM work (in my fork of the mono repo.) 👍

File: cmd/core-data/res/docker/configuration.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ValidateCheck = false
+
+[Service]
+BootTimeout = 30000
+ClientMonitor = 15000
+CheckInterval = '10s'
+Host = 'edgex-core-data'
+Port = 48080
+Protocol = 'http'
+ReadMaxLimit = 100
```
Comment (jduranf) [inline cmd/core-data/res/docker/configuration.toml]: Why there are two ReadMaxLimit? This second one is not used in core-data, and also in core-command

Comment (tsconn23) [reply cmd/core-data/res/docker/configuration.toml]: Good catch. This was an error when I merged a file conflict from another PR.

File: cmd/core-data/res/docker/configuration.toml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+AddToEventQueue = true
+DeviceUpdateLastConnected = false
+FormatSpecifier = '%(\\d=\\$)?([-#= 0(\\<]*)?(\\d=)?(\\.\\d=)?([tT])?([a-zA-Z%])'
```
[the remark this answers is not in the document]
Comment (tsconn23) [reply cmd/core-data/res/docker/configuration.toml]: Will be removed.

File: cmd/core-data/res/docker/configuration.toml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+AddToEventQueue = true
```
[the remark this answers is not in the document]
Comment (tsconn23) [reply cmd/core-data/res/docker/configuration.toml]: Will be removed

**Pull request merged**